internal/provider/provider.go

```diff
@@ -54,20 +54,21 @@ func New() *schema.Provider {
 			},
 		},
 		ResourcesMap: map[string]*schema.Resource{
-			"boundary_account":              resourceAccount(),
-			"boundary_account_password":     resourceAccountPassword(),
-			"boundary_account_oidc":         resourceAccountOidc(),
-			"boundary_auth_method":          resourceAuthMethod(),
-			"boundary_auth_method_password": resourceAuthMethodPassword(),
-			"boundary_auth_method_oidc":     resourceAuthMethodOidc(),
-			"boundary_group":                resourceGroup(),
-			"boundary_host":                 resourceHost(),
-			"boundary_host_catalog":         resourceHostCatalog(),
-			"boundary_host_set":             resourceHostset(),
-			"boundary_role":                 resourceRole(),
-			"boundary_scope":                resourceScope(),
-			"boundary_target":               resourceTarget(),
-			"boundary_user":                 resourceUser(),
+			"boundary_account":                  resourceAccount(),
+			"boundary_account_password":         resourceAccountPassword(),
+			"boundary_account_oidc":             resourceAccountOidc(),
+			"boundary_auth_method":              resourceAuthMethod(),
+			"boundary_auth_method_password":     resourceAuthMethodPassword(),
+			"boundary_auth_method_oidc":         resourceAuthMethodOidc(),
+			"boundary_credential_library_vault": resourceCredentialLibraryVault(),
```

Comment:
Since libraries depend on stores I would have expected to see the work on the store to come first.  The way you have it here you'll need to create the store external to TF and then use TF for library operations.

Reply:
Yeah, I realized this part way through writing the library. I started adding the store this morning so I can finish writing tests for the library resource. They'll both ship in this PR :)